browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/BraveAutocompleteCoordinator.java

```diff
@@ -0,0 +1,188 @@
+                        OmniboxSuggestionUiType.DEFAULT,
+                        parent -> new BaseSuggestionView<View>(
+                                parent.getContext(), R.layout.omnibox_basic_suggestion),
+                        new BaseSuggestionViewBinder<View>(SuggestionViewViewBinder::bind));
+
+                adapter.registerType(
+                        OmniboxSuggestionUiType.EDIT_URL_SUGGESTION,
+                        parent -> new EditUrlSuggestionView(parent.getContext()),
+                        new EditUrlSuggestionViewBinder());
+
+                adapter.registerType(
+                        OmniboxSuggestionUiType.ANSWER_SUGGESTION,
+                        parent -> new BaseSuggestionView<View>(
+                                parent.getContext(), R.layout.omnibox_answer_suggestion),
+                        new BaseSuggestionViewBinder<View>(AnswerSuggestionViewBinder::bind));
+
+                adapter.registerType(
+                        OmniboxSuggestionUiType.ENTITY_SUGGESTION,
+                        parent -> new BaseSuggestionView<View>(
+                                parent.getContext(), R.layout.omnibox_entity_suggestion),
+                        new BaseSuggestionViewBinder<View>(EntitySuggestionViewBinder::bind));
+
+                adapter.registerType(
+                        OmniboxSuggestionUiType.TAIL_SUGGESTION,
+                        parent -> new BaseSuggestionView<TailSuggestionView>(
+                                new TailSuggestionView(parent.getContext())),
+                        new BaseSuggestionViewBinder<TailSuggestionView>(
+                                TailSuggestionViewBinder::bind));
+
+                adapter.registerType(
+                        OmniboxSuggestionUiType.CLIPBOARD_SUGGESTION,
+                        parent -> new BaseSuggestionView<View>(
+                                parent.getContext(), R.layout.omnibox_basic_suggestion),
+                        new BaseSuggestionViewBinder<View>(SuggestionViewViewBinder::bind));
+
+                adapter.registerType(
+                        OmniboxSuggestionUiType.TILE_NAVSUGGEST,
+                        MostVisitedTilesProcessor::createView,
+                        BaseCarouselSuggestionViewBinder::bind);
+
+                adapter.registerType(
+                        BraveOmniboxSuggestionUiType.HEADER,
+                        parent -> new HeaderView(parent.getContext()),
+                        HeaderViewBinder::bind);
+
+                adapter.registerType(
+                        OmniboxSuggestionUiType.PEDAL_SUGGESTION,
+                        parent -> new PedalSuggestionView<View>(
+                                parent.getContext(), R.layout.omnibox_basic_suggestion),
+                        new PedalSuggestionViewBinder<View>(SuggestionViewViewBinder::bind));
+
+                adapter.registerType(BraveOmniboxSuggestionUiType.BRAVE_SEARCH_PROMO_BANNER,
```

Comment:
Apart form this the rest seems to be the copy of the parent. Can we extend `SuggestionListViewHolder` and get there to the `adapter` and register type in `BraveSuggestionListViewHolder`?